### DISCUSSION TITLE
How to align node with CSS grid

### DISCUSSION THREAD

**Discussion by trivedi** [xyflow/xyflow#3320] [Q&A]
Nodes take position using x,y coordinates. But is there a way for me to align them to my CSS grid?

I have a simple 5 column grid:
`grid-template-columns: [column-A-start] repeat(2, 1fr) [coulmn-A-end] 1fr [column-B-start] repeat(2, 1fr) [column-B-end]`

I want some nodes in column A and some nodes in column B. The edges connect nodes in A to B (bidirectionally, so B to A as well).
The edges will be in the middle column.

**Comment by ToJen** [comment]
Any luck?

**Reply by trivedi** [reply]
As far as I can tell I don't think it's possible

**Comment by mayurjadhav2002** [comment]
I use conditional rendering to add inline flex styles to my nodes. it only applies to the child node and I am calling a recursive function in my parent node. 
```jsx
<div className=' relative h-[760px] flex gap-40 mx-auto justify-between'>
          {data.child.map((childData, index) => (
            <ChildNode key={index} data={childData} />
          ))}
        </div>
```

Instead of using the reactflow given structure of nodes, I made hierarchical json object of nodes. then map them into dom.
I am trying to do this without hierarchical values.
